pyjobshop/ProblemData.py

```diff
@@ -2,13 +2,66 @@
 from copy import deepcopy
 from dataclasses import dataclass, field, fields
 from itertools import pairwise
-from typing import Sequence, TypeVar
+from typing import Protocol, Sequence, TypeAlias, TypeVar
 
 from pyjobshop.constants import MAX_VALUE
+from pyjobshop.protocols import DataclassInstance, SizedDataclassInstance
 
 _T = TypeVar("_T")
 
+MISSING: Sequence = ()
 
+Break: TypeAlias = tuple[int, int]
+Breaks: TypeAlias = Sequence[Break]
+
+
+class HasPostInit(Protocol):
+    """
+    Specify the presence of a ``__post_init__()`` method in a class (or
+    indeed in a superclass).  Used for dataclass validation in the mixins
+    below.
+    """
+
+    def __post_init__(self): ...
+
+
+class CheckCapacityMixin(HasPostInit, Protocol):
+    """
+    Specify the presence of a ``capacity`` field, and provide a validation
+    check for it.
+    """
+
+    capacity: int
+
+    def __post_init__(self):
+        super().__post_init__()
+        if self.capacity < 0:
+            raise ValueError("Capacity must be non-negative.")
```

Comment:
For checking capacity, I don't think it's necessary to create a helper. It's a simple 2-line statement in two places, so that's just 4 lines of code, as opposed to the 10 lines needed for this mixin.